src/storage/Conditions.ts

```diff
@@ -23,13 +24,29 @@ export interface Conditions {
 
   /**
    * Checks validity based on the given metadata.
-   * @param metadata - Metadata of the representation.
+   * @param metadata - Metadata of the representation. Undefined if the resource does not exist.
    */
-  matchesMetadata: (metadata: RepresentationMetadata) => boolean;
+  matchesMetadata: (metadata?: RepresentationMetadata) => boolean;
   /**
-   * Checks validity based on the given ETag and/org date.
+   * Checks validity based on the given ETag and/or date.
+   * This function assumes the resource being checked exists.
+   * If not, the `matchesMetadata` function should be used.
    * @param eTag - Condition based on ETag.
    * @param lastModified - Condition based on last modified date.
    */
   matches: (eTag?: string, lastModified?: Date) => boolean;
 }
+
+/**
+ * Generates an ETag based on the last modified date of a resource.
+ * @param metadata - Metadata of the resource.
+ *
+ * @returns the generated ETag. Undefined if no last modified date was found.
+ */
+export function getETag(metadata: RepresentationMetadata): string | undefined {
+  const modified = metadata.get(DC.terms.modified);
+  if (modified) {
+    const date = new Date(modified.value);
+    return `"${date.getTime()}"`;
```

Comment:
Perhaps hash this value?

Reply:
I thought about this, but didn't really see an advantage compared to just using the timestamp? Unless perhaps for privacy reasons?